inventory_management_system_api/services/utils.py

```diff
@@ -35,80 +35,72 @@ def generate_code(name: str, entity_type: str) -> str:
-    supplied_properties_dict = _merge_non_mandatory_catalogue_item_properties(
-        defined_properties_dict, supplied_properties_dict
-    )
-    # Supplied catalogue item properties do not have names as we can't trust they would be correct
-    _add_catalogue_item_property_names(defined_properties_dict, supplied_properties_dict)
-    # Supplied catalogue item properties do not have units as we can't trust they would be correct
-    _add_catalogue_item_property_units(defined_properties_dict, supplied_properties_dict)
-    # The values of the supplied catalogue item properties may not be of the expected types
-    _validate_catalogue_item_property_values(defined_properties_dict, supplied_properties_dict)
+    defined_properties_dict = _create_properties_dict(defined_properties)
+    supplied_properties_dict = _create_properties_dict(supplied_properties)
+
+    # Some mandatory  may not have been supplied
+    _check_missing_mandatory_properties(defined_properties_dict, supplied_properties_dict)
+    # Some non-mandatory  may not have been supplied
+    supplied_properties_dict = _merge_non_mandatory_properties(defined_properties_dict, supplied_properties_dict)
+    # Supplied  do not have names as we can't trust they would be correct
+    _add_property_names(defined_properties_dict, supplied_properties_dict)
+    # Supplied  do not have units as we can't trust they would be correct
+    _add_property_units(defined_properties_dict, supplied_properties_dict)
+    # The values of the supplied  may not be of the expected types
+    _validate_property_values(defined_properties_dict, supplied_properties_dict)
 
     return list(supplied_properties_dict.values())
 
 
-def _create_catalogue_item_properties_dict(
-    catalogue_item_properties: Union[List[CatalogueItemPropertyOut], List[PropertyPostRequestSchema]]
+def _create_properties_dict(
+    properties: Union[List[CatalogueCategoryPropertyOut], List[PropertyPostRequestSchema]]
 ) -> Dict[str, Dict]:
     """
-    Convert a list of catalogue item property objects into a dictionary where the keys are the catalogue item
-    property IDs and the values are the catalogue item property dictionaries.
+    Convert a list of property objects into a dictionary where the keys are the catalogue item
+    property IDs and the values are the property dictionaries.
 
-    :param catalogue_item_properties: The list of catalogue item property objects.
-    :return: A dictionary where the keys are the catalogue item property IDs and the values are the catalogue item
+    :param properties: The list of property objects.
+    :return: A dictionary where the keys are the property IDs and the values are the catalogue item
         property dictionaries.
     """
-    return {
-        catalogue_item_property.id: catalogue_item_property.model_dump()
-        for catalogue_item_property in catalogue_item_properties
-    }
+    return {property.id: property.model_dump() for property in properties}
```

Comment:
Shadows the built-in `property` name.
```suggestion
    return {prop.id: prop.model_dump() for prop in properties}
```